confgenerator/logging_receivers.go

```diff
@@ -272,3 +272,28 @@ func (r LoggingReceiverSystemd) Components(tag string) []fluentbit.Component {
 func init() {
 	LoggingReceiverTypes.RegisterType(func() Component { return &LoggingReceiverSystemd{} }, "linux")
 }
+
+// A LoggingReceiverMetric represents the user configuration for a fluentbit self-metric receiver.
+type LoggingReceiverMetric struct {
+	ConfigComponent `yaml:",inline"`
+}
+
+func (r LoggingReceiverMetric) Components(tag string) []fluentbit.Component {
+	return []fluentbit.Component{{
+		Kind: "INPUT",
+		Config: map[string]string{
+			// https://docs.fluentbit.io/manual/pipeline/inputs/systemd
+			"Name":            "fluentbit_metrics",
```

Comment:
Similar to before, we do not want to expose the implementation details of Fluent Bit to users. Instead, we make the self metrics enabled by default and users can not turn it off. This way we guarantee:

* Ops Agent self metrics are always ingested 
* In the future, we have flexibility to swap the underlying agents (e.g. Fluent Bit or Open Telemetry Collector) if needed.

Reply:
Changed implementation to hardcode the data as discussed